nifi-commons/nifi-security-utils/src/main/java/org/apache/nifi/security/util/crypto/Argon2SecureHasher.java

```diff
@@ -53,11 +53,14 @@
     private final int iterations;
     private final int saltLength;
 
-    private final boolean usingStaticSalt;
+    private boolean usingStaticSalt;
 
     // A 16 byte salt (nonce) is recommended for password hashing
     private static final byte[] staticSalt = "NiFi Static Salt".getBytes(StandardCharsets.UTF_8);
 
+    // Upper boundary for several cost parameters
+    private static final double upperBoundary = Math.pow(2, 32) - 1;
```

Comment:
Should be `int` rather than `double`.

Reply:
As `2^32 - 1` is `4*10^12`, an `int` can hold `2^32` possible values, but _signed_, so the `Integer.MAX_VALUE` is `2....*10^12`. A `long` is required for all of these fields.

Reply:
Used `Integer` to hold unsigned ints.